### PR TITLE
GAA: Send message before redirecting

### PR DESCRIPTION
**Pull request by ChrisAntaki**
This PR ensures we send a postMessage before redirecting the page

### CODE REVIEW

File: src/utils/gaa-test.js [hunk 1]
```diff
@@ -1979,13 +1979,21 @@ describes.realWin('GaaGoogle3pSignInButton', {}, () => {
...
+      // Wait for promises and intervals to resolve.
+      clock.tick(100);
+      await tick(10);
+
       // Click button.
       self.document.getElementById(GOOGLE_3P_SIGN_IN_BUTTON_ID).click();
       clock.tick(100);
       await tick(10);
 
-      expect(self.open).to.have.been.calledOnce;
```
Comment (ChrisAntaki) [inline src/utils/gaa-test.js:1987]: This expectation seemed redundant, since subsequent tests specifically expect this to be called in certain ways

Comment (ChrisAntaki) [reply src/utils/gaa-test.js:1987]: Also this expectation didn't seem to match the test description

File: src/utils/gaa-test.js [hunk 2]
```diff
@@ -1979,13 +1979,21 @@ describes.realWin('GaaGoogle3pSignInButton', {}, () => {
       clock.tick(100);
       await tick(10);
 
+      // Send intro post message.
```
Comment (ChrisAntaki) [inline src/utils/gaa-test.js:1982]: This test code lets `sendMessageToParentFnPromise` resolve

File: src/utils/gaa-test.js [hunk 3]
```diff
@@ -2113,16 +2130,54 @@ describes.realWin('GaaGoogle3pSignInButton', {}, () => {
...
+      // Wait for `open` to be called.
+      await new Promise((resolve) => void self.open.callsFake(resolve));
+
       expect(self.open).to.have.been.calledWithExactly(
         GOOGLE_3P_AUTH_URL,
         '_parent'
       );
     });
+
+    it('should open window twice when redirectMode is true', async () => {
```
Comment (ChrisAntaki) [inline src/utils/gaa-test.js:2154]: Seems worth explicitly testing the second attempt to open a window

**Review by jpettitt — APPROVED** [review]
Approved, may be a good safety strategy to change the code then test test as a second PR, that ways it's clear the code still passes the tests, simultaneous changes make it easy to introduce bugs the tests don't catch.

**Pull request merged**